src/data.py

```diff
@@ -424,7 +424,10 @@ def optimize_limits(self) -> None:
                 xydata = numpy.asarray(
                     item_.ydata if index % 2 else item_.xdata,
                 )
-                xydata = xydata[numpy.isfinite(xydata)]
+                try:
+                    xydata = xydata[numpy.isfinite(xydata)]
```

Comment:
This does not always work with certain exceptions, better to just skip optimizing limits in those cases.